.github/workflows/main.yml

```diff
@@ -1,87 +1,86 @@
-name: Main
+name: Code quality
 
 on:
   push:
     branches:
-      - main
       - master
   pull_request:
     branches:
-      - main
       - master
+  workflow_dispatch:
 
 jobs:
   lint:
-    name: ESLint
-    runs-on: ${{ matrix.os }}
-
-    strategy:
-      matrix:
-        os: [ubuntu-latest]
-        node: [14]
+    name: Lint
+    runs-on: ubuntu-latest
 
     steps:
       - name: Checkout
         uses: actions/checkout@master
+        with:
+          fetch-depth: 1
+          clean: true
+          ref: 'master'
 
-      - name: Setup node env
+      - name: Configure node environment
         uses: actions/setup-node@v2.1.4
         with:
-          node-version: ${{ matrix.node }}
+          node-version: 14
 
       - name: Get yarn cache directory path
         id: yarn-cache-dir-path
         run: echo "::set-output name=dir::$(yarn cache dir)"
 
       - name: Cache node_modules
         uses: actions/cache@v2.1.4
-        id: yarn-cache # use this to check for `cache-hit` (`steps.yarn-cache.outputs.cache-hit != 'true'`)
+        id: yarn-cache
         with:
           path: ${{ steps.yarn-cache-dir-path.outputs.dir }}
           key: ${{ runner.os }}-yarn-${{ hashFiles('**/yarn.lock') }}
           restore-keys: |
             ${{ runner.os }}-yarn-
 
       - name: Install dependencies
-        run: yarn
+        run: yarn install --frozen-lockfile
 
       - name: Run linter
         run: yarn lint
-  jest:
-    name: Jest
-    runs-on: ${{ matrix.os }}
 
-    strategy:
-      matrix:
-        os: [ubuntu-latest]
-        node: [14]
+  test:
+    name: Test
+    runs-on: ubuntu-latest
 
     steps:
       - name: Checkout
         uses: actions/checkout@master
+        with:
+          fetch-depth: 1
+          clean: true
+          ref: 'master'
 
-      - name: Setup node env
+      - name: Configure node environment
         uses: actions/setup-node@v2.1.4
         with:
-          node-version: ${{ matrix.node }}
+          node-version: 14
+
+      - name: Install dependencies
```

Comment:
I believe this should be done after the cache step, since that step restores the cache from the previous run (if it matches)